LayoutTests/editing/selection/ios/hide-selection-in-hidden-container-noneditable.html

```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html> <!-- webkit-test-runner [ useFlexibleViewport=true ] -->
+<html>
+<head>
+<meta name="viewport" content="width=device-width, user-scalable=no">
+<script src="../../../resources/ui-helper.js"></script>
+<style>
+body, html {
+    width: 100%;
+    height: 100%;
+    margin: 0;
+}
+
+.hidden-container {
+    width: 320px;
+    height: 320px;
+    border: solid 2px silver;
+    box-sizing: border-box;
+    overflow: hidden;
+}
+
+</style>
+</head>
+<body>
+<div class="hidden-container">
+</div>
+<p id="text">Hello</p>
+
+<script>
+(async () => {
+    if (!window.testRunner)
+        return;
+
+    testRunner.dumpAsText();
+    testRunner.waitUntilDone();
+
+    console.log("1. Long press to select.");
+    let pText = document.querySelector("#text");
+    await UIHelper.longPressElement(pText);
+
+    console.log("2. Wait for selection to appear.");
+    await UIHelper.waitForSelectionToAppear();
+
+    console.log("3. Move element to hidden container.");
+    let hiddenDiv = document.querySelector(".hidden-container");
+    hiddenDiv.append(pText);
+
+    console.log("4. Wait for selection to disappear.");
+    await UIHelper.waitForSelectionToDisappear();
```

Comment:
Could we double check that this test fails without the fix in this patch?

Reply:
Added the `shouldBe` to the test, but it passes without the fix... Is there a way to "debug" the test when running (e.g. breakpoints, printing)?

Reply:
> Added the shouldBe to the test, but it passes without the fix.

Oh, my bad — reparenting the selection container _collapses_ the DOM selection rather than clearing it altogether, so the `rangeCount` will still be equal to 1. Can we instead check this?

```
shouldBeEqualToString("getSelection().type", "Range");
shouldBeEqualToString("getSelection().toString()", "Hello");
```

This should confirm that the contents of `pText` remain selected, after making it hidden.

> Is there a way to "debug" the test when running (e.g. breakpoints, printing)

Yep, both are viable approaches for debugging tests (e.g. `WTFLogAlways`, or using `lldb -n WebKitTestRunnerApp -w` or `lldb -n com.apple.WebKit.WebContent.Development -w`).